### PR TITLE
SER-506 Image Review tests

### PR DESCRIPTION
**Pull request by nikodamn**
* Basic tests for Image Review UI.

Ticket: SER-506

Reviewers:
@Wikia/services-team @ludwikkazmierczak 

Dependencies:
* Selenium-config changes required

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/ImageReview.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    return actionButtons.size();
+  }
+
+  public boolean areNavbarButtonsClickable() {
+    wait.forElementVisible(navbar);
+
+    for (WebElement navbarButton : navbarButtons) {
+      try {
+        wait.forElementClickable(navbarButton);
+      } catch(TimeoutException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/ImageReview.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import com.wikia.webdriver.elements.mercury.pages.ImageReview;
+
+import org.testng.annotations.Test;
+
+@Test(groups = "ImageReview")
+public class ImageReviewTests extends NewTestTemplate {
+
+  private final int STAFF_NAVBAR_BUTTONS_NUMBER = 2;
+  private final int REVIEWER_NAVBAR_BUTTONS_NUMBER = 1;
+  private final int STAFF_ACTION_BUTTONS_NUMBER = 4;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: MAJOR Make this final field static too. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)
MAJOR Rename this field "STAFF_ACTION_BUTTONS_NUMBER" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  private final int REVIEWER_ACTION_BUTTONS_NUMBER = 1;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: MAJOR Make this final field static too. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)
MAJOR Rename this field "REVIEWER_ACTION_BUTTONS_NUMBER" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  private final String ANON_ALERT_MESSAGE = "Unable to load image review\n×";
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: MAJOR Rename this field "ANON_ALERT_MESSAGE" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)
MAJOR Make this final field static too. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.elements.mercury.pages.ImageReview;
+
+import org.testng.annotations.Test;
+
+@Test(groups = "ImageReview")
+public class ImageReviewTests extends NewTestTemplate {
+
+  private final int STAFF_NAVBAR_BUTTONS_NUMBER = 2;
+  private final int REVIEWER_NAVBAR_BUTTONS_NUMBER = 1;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: MAJOR Rename this field "REVIEWER_NAVBAR_BUTTONS_NUMBER" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)
MAJOR Make this final field static too. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java [hunk 6]
```diff
@@ -103,6 +104,22 @@ public String getUrlForWiki(String wikiName, boolean addWWW) {
...
 
+  public  String getWikiGlobalURL(){
+    EnvType env = Configuration.getEnvType(this.env);
+
+    switch (env) {
+      case DEV: {
+        String devBoxOwner = this.env.split("-")[1];
+        return String.format("http://%s.www.%s", devBoxOwner, env.getWikiaDomain());
+      }
+      case SANDBOX: {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR Extract this nested code block into a method. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1199)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java [hunk 7]
```diff
@@ -120,6 +121,21 @@ protected boolean isElementOnPage(WebElement element) {
...
   /**
+   * Method to check if WebElement is displayed on the page
+   * @param element
+   * @return true if element is displayed, otherwise return false
+   */
+
+  protected boolean isElementDisplayed(WebElement element) {
+    try {
+      return element.isDisplayed();
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues:
* MAJOR 17 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/ImageReview.java (hunk 1 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/ImageReview.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java (hunk 2 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: MAJOR Rename this field "STAFF_ACTION_BUTTONS_NUMBER" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)
MAJOR Make this final field static too. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java (hunk 3 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: MAJOR Rename this field "REVIEWER_ACTION_BUTTONS_NUMBER" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)
MAJOR Make this final field static too. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java (hunk 4 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: MAJOR Rename this field "ANON_ALERT_MESSAGE" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)
MAJOR Make this final field static too. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+  private final int STAFF_ACTION_BUTTONS_NUMBER = 4;
+  private final int REVIEWER_ACTION_BUTTONS_NUMBER = 1;
+  private final String ANON_ALERT_MESSAGE = "Unable to load image review\n×";
+  private final String UNREVIEWED_VIEW_MESSAGE = "UNREVIEWED";
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: MAJOR Rename this field "UNREVIEWED_VIEW_MESSAGE" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)
MAJOR Make this final field static too. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+import com.wikia.webdriver.common.core.helpers.User;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.elements.mercury.pages.ImageReview;
+
+import org.testng.annotations.Test;
+
+@Test(groups = "ImageReview")
+public class ImageReviewTests extends NewTestTemplate {
+
+  private final int STAFF_NAVBAR_BUTTONS_NUMBER = 2;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: MAJOR Rename this field "STAFF_NAVBAR_BUTTONS_NUMBER" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)
MAJOR Make this final field static too. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java (hunk 5 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: MAJOR Rename this field "REVIEWER_NAVBAR_BUTTONS_NUMBER" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)
MAJOR Make this final field static too. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java (hunk 6 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR Extract this nested code block into a method. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1199)

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java [hunk 10]
```diff
@@ -103,6 +104,22 @@ public String getUrlForWiki(String wikiName, boolean addWWW) {
     }
   }
 
+  public  String getWikiGlobalURL(){
+    EnvType env = Configuration.getEnvType(this.env);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR Rename "env" which hides the field declared at line 14. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AHiddenFieldCheck)

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java [hunk 11]
```diff
@@ -103,6 +104,22 @@ public String getUrlForWiki(String wikiName, boolean addWWW) {
...
+
+    switch (env) {
+      case DEV: {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR Extract this nested code block into a method. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1199)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java (hunk 7 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues:
* MAJOR 17 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java (hunk 10 above)
Comment (sqreek) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: +1

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java [hunk 12]
```diff
@@ -218,7 +234,7 @@ public String getCurrentUrl() {
     return driver.getCurrentUrl();
   }
 
-  public void getUrl(String url) {
+  public void   getUrl(String url) {
```
Comment (sqreek) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: ?

File: src/test/java/com/wikia/webdriver/testcases/discussions/ReportingPostTests.java [hunk 13]
```diff
@@ -202,7 +202,6 @@ public void anonUserOnDesktopCanNotSeeDeletedPostOnPostsListPage() {
 
   @Test(groups = "discussions-anonUserDesktopReporting")
   @Execute(asUser = User.ANONYMOUS)
-  @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
```
Comment (sqreek) [inline src/test/java/com/wikia/webdriver/testcases/discussions/ReportingPostTests.java]: is those changes are required to this ticket?

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java (hunk 9 above)
Comment (sqreek) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: 👍

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java (hunk 8 above)
Comment (sqreek) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: Missing statics

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+  @Test(groups = "anonCannotEnterImageReview")
+  public void anonCannotEnterImageReview() {
+    ImageReview imageReviewPage = new ImageReview().open();
+
+    Assertion.assertTrue(imageReviewPage.isAlertNotificationDisplayed());
+    Assertion.assertEquals(imageReviewPage.getAlertNotificationText(), ANON_ALERT_MESSAGE);
+  }
+
+  @Execute(asUser = User.USER)
+  @Test(groups = "regularUserCannotEnterImageReview")
```
Comment (sqreek) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/ImageReviewTests.java]: groups?

**Comment by sqreek** [comment]
overall lg. 
I recommend to not mix checks and methods in https://github.com/Wikia/selenium-tests/pull/1622/files#diff-213182055224ace0d5ab5f42bb6cc862 and put checks near each and methods in the end

File: src/test/java/com/wikia/webdriver/testcases/discussions/ReportingPostTests.java (hunk 13 above)
Comment (nikodamn) [inline src/test/java/com/wikia/webdriver/testcases/discussions/ReportingPostTests.java]: Of course not. Great catch! :+1:

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues:
* MAJOR 17 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java [hunk 15]
```diff
@@ -76,33 +77,49 @@ public String getUrlForWiki(boolean addWWW) {
...
     final String wikiaName = getWikiaGlobalName(wikiName);
 
     String www = "";
     if (addWWW) {
       www = "www.";
     }
 
-    switch (env) {
+    switch (envType) {
       case DEV: {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR Extract this nested code block into a method. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1199)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 4 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/ImageReview.java [hunk 16]
```diff
@@ -0,0 +1,169 @@
...
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/ImageReview.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java [hunk 17]
```diff
@@ -76,33 +77,49 @@ public String getUrlForWiki(boolean addWWW) {
...
 
+  public String getWikiGlobalURL(){
+    EnvType env = Configuration.getEnvType(this.env);
+
+    switch (env) {
+      case DEV: {
+        String devBoxOwner = this.env.split("-")[1];
+        return String.format("http://%s.www.%s", devBoxOwner, env.getWikiaDomain());
+      }
+      case SANDBOX: {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR Extract this nested code block into a method. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1199)

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java (hunk 15 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR Extract this nested code block into a method. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1199)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java (hunk 7 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 4 major

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 4 major

Watch the comments in this conversation to review them.

**Pull request merged**